master/buildbot/test/unit/test_steps_diffinfo.py

```diff
@@ -0,0 +1,132 @@
+        self.expect_log_file_stderr('stdio-merge-base', 'fatal: Not a valid object name')
+        self.expectOutcome(result=results.FAILURE, state_string="diffinfo (failure)")
+        return self.runStep()
+
+    def test_diff_failure(self):
+        self.setupStep(diffinfo.DiffInfo())
+        self.expectCommands(
+            ExpectShell(workdir='wkdir', command=['git', 'merge-base', 'HEAD', 'master'])
+            + Expect.log('stdio-merge-base', stdout='1234123412341234')
+            + 0,
+            ExpectShell(workdir='wkdir', command=['git', 'diff', '-U0', '1234123412341234', 'HEAD'])
+            + Expect.log('stdio-diff', stderr='fatal: ambiguous argument')
+            + 1,
+            )
+        self.expectLogfile('stdio-merge-base', '1234123412341234')
+        self.expect_log_file_stderr('stdio-diff', 'fatal: ambiguous argument')
+        self.expectOutcome(result=results.FAILURE, state_string="diffinfo (failure)")
+        return self.runStep()
+
+    def test_empty_diff(self):
+        self.setupStep(diffinfo.DiffInfo())
+        self.expectCommands(
+            ExpectShell(workdir='wkdir', command=['git', 'merge-base', 'HEAD', 'master'])
+            + Expect.log('stdio-merge-base', stdout='1234123412341234')
+            + 0,
+            ExpectShell(workdir='wkdir', command=['git', 'diff', '-U0', '1234123412341234', 'HEAD'])
+            + Expect.log('stdio-diff', stdout='')
+            + 0,
+            )
+        self.expectLogfile('stdio-merge-base', '1234123412341234')
+        self.expect_log_file_stderr('stdio-diff', '')
+        self.expectOutcome(result=results.SUCCESS, state_string="diffinfo")
+        self.expect_build_data('diffinfo-master', b'[]', 'DiffInfo')
+        return self.runStep()
+
+    def test_complex_diff(self):
+        self.setupStep(diffinfo.DiffInfo())
+        self.expectCommands(
+            ExpectShell(workdir='wkdir', command=['git', 'merge-base', 'HEAD', 'master'])
+            + Expect.log('stdio-merge-base', stdout='1234123412341234')
+            + 0,
+            ExpectShell(workdir='wkdir', command=['git', 'diff', '-U0', '1234123412341234', 'HEAD'])
+            + Expect.log('stdio-diff', stdout='''\
+diff --git a/file1 b/file1
+deleted file mode 100644
+index 42f90fd..0000000
+--- a/file1
++++ /dev/null
+@@ -1,3 +0,0 @@
+-line11
+-line12
+-line13
+diff --git a/file2 b/file2
+index c337bf1..1cb02b9 100644
+--- a/file2
++++ b/file2
+@@ -4,0 +5,3 @@ line24
++line24n
++line24n2
++line24n3
+@@ -15,0 +19,3 @@ line215
++line215n
++line215n2
++line215n3
+diff --git a/file3 b/file3
+new file mode 100644
+index 0000000..632e269
+--- /dev/null
++++ b/file3
+@@ -0,0 +1,3 @@
++line31
++line32
++line33
+''')
+            + 0,
+            )
+        self.expectLogfile('stdio-merge-base', '1234123412341234')
+        self.expectOutcome(result=results.SUCCESS, state_string="diffinfo")
+
+        diff_info = (
+            b'[{"source_file": "a/file1", "target_file": "/dev/null", ' +
```

Comment:
I would strip the 'a/' which is git specific.

Reply:
Added `--no-prefix` option to git to get rid of these.